libgraph/src/analytics/k_shortest_paths/sssp.cpp

```diff
@@ -0,0 +1,462 @@
+      graph.template GetData<NodeCount>(n) = 0;
+      for (auto e : graph.OutEdges(n)) {
+        edge_data[e] = graph.template GetEdgeData<EdgeWeight>(e);
+      }
+    });
+
+    katana::StatTimer execTime("SSSP");
+    execTime.start();
+
+    katana::InsertBag<std::pair<Weight, Path*>> paths;
+    katana::InsertBag<Path*> path_pointers;
+
+    bool reachable = true;
+
+    switch (algoReachability) {
+    case async:
+      reachable = CheckReachabilityAsync<BFSUpdateRequest>(
+          &graph, source, BFSReqPushWrap(), BFSOutEdgeRangeFn{&graph},
+          reportNode);
+      break;
+    case syncLevel:
+      reachable = CheckReachabilitySync(&graph, source, reportNode);
+      break;
+    default:
+      std::abort();
+    }
+
+    PathAlloc path_alloc;
+
+    if (reachable) {
+      switch (plan.algorithm()) {
+      case SsspPlan::kDeltaTile:
+        DeltaStepAlgo<SSSPSrcEdgeTile, OBIM>(
+            &edge_data, &graph, source, SSSPSrcEdgeTilePushWrap{&graph},
+            SSSPTileRangeFn(), &paths, &path_pointers, path_alloc, reportNode,
+            numPaths, stepShift);
+        break;
+      case SsspPlan::kDeltaStep:
+        DeltaStepAlgo<SSSPUpdateRequest, OBIM>(
+            &edge_data, &graph, source, SSSPReqPushWrap(),
+            SSSPOutEdgeRangeFn{&graph}, &paths, &path_pointers, path_alloc,
+            reportNode, numPaths, stepShift);
+        break;
+      case SsspPlan::kDeltaStepBarrier:
+        katana::gInfo("Using OBIM with barrier\n");
+        DeltaStepAlgo<SSSPUpdateRequest, OBIM_Barrier>(
+            &edge_data, &graph, source, SSSPReqPushWrap(),
+            SSSPOutEdgeRangeFn{&graph}, &paths, &path_pointers, path_alloc,
+            reportNode, numPaths, stepShift);
+        break;
+
+      default:
+        return katana::ErrorCode::InvalidArgument;
+      }
+    }
+
+    execTime.stop();
+    page_alloc.Report();
+
+    if (reachable) {
+      std::multimap<uint32_t, Path*> paths_map;
+
+      for (auto pair : paths) {
+        paths_map.insert(std::make_pair(pair.first, pair.second));
+      }
+
+      katana::gPrint("Node ", report, " has these k paths:\n");
+
+      uint32_t num =
+          (paths_map.size() > numPaths) ? numPaths : paths_map.size();
+
+      auto it_report = paths_map.begin();
+
+      for (uint32_t iter = 0; iter < num; iter++) {
+        const Path* path = it_report->second;
+        PrintPath(path);
+        katana::gPrint(" ", report, "\n");
+        katana::gPrint("Weight: ", it_report->first, "\n");
+        it_report++;
+      }
+
+      katana::do_all(katana::iterate(path_pointers), [&](Path* p) {
+        path_alloc.DeletePath(p);
+      });
+    }
+
+    return katana::ResultSuccess();
+  }
+};
+
+template <typename GraphTy, typename Weight>
+katana::Result<void>
+Ksp(GraphTy& pg, unsigned int startNode, unsigned int reportNode,
+    AlgoReachability algoReachability, unsigned int numPaths,
+    unsigned int stepShift, SsspPlan plan) {
```

Comment:
same as above; rename vars and add doxygen comments

Reply:
@dqawami Please use the snake_case for argument (see other apps). Please follow the https://google.github.io/styleguide/cppguide.html